### PR TITLE
add hPutBuilderLen which returns the number of bytes

### PR DESCRIPTION
**Pull request by fumieval**

### CODE REVIEW

File: Data/ByteString/FastBuilder/Internal.hs [hunk 1]
```diff
@@ -715,26 +723,26 @@ growBuffer !bufRef !req = do
...
-  Queue fptr _ <- io $ readIORef qRef
+  Queue fptr _ total <- io $ readIORef qRef
   let !base = unsafeForeignPtrToPtr fptr
   let !cap = end `minusPtr` base
   newFptr <- if minSize <= cap
     then return fptr
     else io $ mallocForeignPtrBytes minSize
   let !newBase = unsafeForeignPtrToPtr newFptr
-  io $ writeIORef qRef $ Queue newFptr 0
+  io $ writeIORef qRef $ Queue newFptr 0 (total + adv)
```
Comment (takano-akio) [inline Data/ByteString/FastBuilder/Internal.hs]: It seems wrong to update the `total` field in both `flushQueue` and `switchQueue`: they are both called in `getBytes_`.

File: Data/ByteString/FastBuilder/Internal.hs [hunk 2]
```diff
@@ -153,7 +154,7 @@ data DynamicSink
       -- to a 'ThreadedSink'.
 
 -- | A mutable buffer. The 'Int' specifies where the data start.
-data Queue = Queue !(ForeignPtr Word8) !Int{-start-}
+data Queue = Queue !(ForeignPtr Word8) !Int{-start-} !Int{-total-}
```
Comment (takano-akio) [inline Data/ByteString/FastBuilder/Internal.hs]: Perhaps it's time to change this datatype into a record? Also please update the comment.

File: Data/ByteString/FastBuilder/Internal.hs [hunk 3]
```diff
@@ -462,19 +463,26 @@ toStrictByteString builder = unsafePerformIO $ do
 
 -- | Output a 'Builder' to a 'IO.Handle'.
 hPutBuilder :: IO.Handle -> Builder -> IO ()
-hPutBuilder !h builder = hPutBuilderWith h 100 4096 builder
+hPutBuilder !h builder = void $ hPutBuilderLen h builder
+{-# INLINE hPutBuilder #-}
+
+-- | Output a 'Builder' to a 'IO.Handle'.
```
Comment (takano-akio) [inline Data/ByteString/FastBuilder/Internal.hs]: "Returns the number of bytes written."

**Review by takano-akio — CHANGES_REQUESTED** [review]
There is a CI failure.

File: Data/ByteString/FastBuilder/Internal.hs (hunk 3 above)
Comment (fumieval) [reply Data/ByteString/FastBuilder/Internal.hs]: done

File: Data/ByteString/FastBuilder/Internal.hs (hunk 2 above)
Comment (fumieval) [reply Data/ByteString/FastBuilder/Internal.hs]: sure

File: Data/ByteString/FastBuilder/Internal.hs (hunk 1 above)
Comment (fumieval) [reply Data/ByteString/FastBuilder/Internal.hs]: One in switchQueue seems to be a mistake. Fixed

File: Data/ByteString/FastBuilder/Internal.hs [hunk 4]
```diff
@@ -462,19 +467,26 @@ toStrictByteString builder = unsafePerformIO $ do
 
 -- | Output a 'Builder' to a 'IO.Handle'.
 hPutBuilder :: IO.Handle -> Builder -> IO ()
-hPutBuilder !h builder = hPutBuilderWith h 100 4096 builder
+hPutBuilder !h builder = void $ hPutBuilderLen h builder
+{-# INLINE hPutBuilder #-}
+
+-- | Output a 'Builder' to a 'IO.Handle'. Returns the number of bytes written.
+hPutBuilderLen :: IO.Handle -> Builder -> IO Int
+hPutBuilderLen !h builder = hPutBuilderWith h 100 4096 builder
```
Comment (fumieval) [inline Data/ByteString/FastBuilder/Internal.hs:475]: Oops I forgot to add INLINE here.

Comment (takano-akio) [reply Data/ByteString/FastBuilder/Internal.hs:475]: I don't think it's a problem. The old version of `hPutBuilder` didn't have an INLINE pragma either.

**Comment by takano-akio** [comment]
Published 0.1.1.0 on Hackage. Thanks!

**Pull request merged**